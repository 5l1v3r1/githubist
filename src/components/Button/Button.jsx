// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import styles from './Button.scss';

type Props = {
  children: Node,
  type: 'button' | 'submit' | 'reset',
  primary?: boolean,
  secondary?: boolean,
  small?: boolean,
  className?: string,
  disabled?: boolean,
};

const Button = ({ children, type, primary, secondary, small, className, disabled }: Props) => {
  const classNames = cx(
    styles.button,
    {
      [styles.primary]: primary,
      [styles.secondary]: secondary,
      [styles.small]: small,
    },
    className,
  );

  return (
    <button type={type} className={classNames} disabled={disabled}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  primary: false,
  secondary: false,
  small: false,
  className: '',
  disabled: false,
};

export default Button;
