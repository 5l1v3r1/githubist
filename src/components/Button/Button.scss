@import '../../theme/theme';

.button {
  padding: px-to-rem(13px) px-to-rem(26px);
  border: 0;
  border-radius: px-to-rem(2px);
  cursor: pointer;
  font-size: px-to-rem(14px);
  letter-spacing: 0.05556rem;
  transition-duration: 0.3s;
  transition-property: background-color, color, border-color;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.7;
  }
}

.small {
  padding: px-to-rem(7px) px-to-rem(13px);
  font-size: px-to-rem(12px);
}

.primary {
  background-color: $primary-color;
  color: $white;

  &:hover {
    background-color: darken($primary-color, 10%);

    &:disabled {
      background-color: $primary-color;
    }
  }
}

.secondary {
  background-color: $secondary-color;
  color: $white;

  &:hover {
    background-color: darken($secondary-color, 10%);

    &:disabled {
      background-color: $secondary-color;
    }
  }
}
