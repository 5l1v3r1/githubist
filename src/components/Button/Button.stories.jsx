// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';

storiesOf('Button', module)
  .add('primary', () => (
    <Button type="button" primary>
      Primary Button
    </Button>
  ))
  .add('secondary', () => (
    <Button type="button" secondary>
      Secondary Button
    </Button>
  ))
  .add('small', () => (
    <Button type="button" primary small>
      Small Button
    </Button>
  ))
  .add('disabled primary', () => (
    <Button type="button" primary disabled>
      Primary Button
    </Button>
  ))
  .add('disabled secondary', () => (
    <Button type="button" secondary disabled>
      Secondary Button
    </Button>
  ));
