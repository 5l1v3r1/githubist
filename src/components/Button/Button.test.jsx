// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

it('renders correctly primary button', () => {
  const wrapper = shallow(
    <Button type="button" primary>
      Button
    </Button>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly secondary button', () => {
  const wrapper = shallow(
    <Button type="button" secondary>
      Button
    </Button>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly small button', () => {
  const wrapper = shallow(
    <Button type="button" small>
      Button
    </Button>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly disabled button', () => {
  const wrapper = shallow(
    <Button type="button" disabled>
      Button
    </Button>,
  );

  expect(wrapper.find('button').prop('disabled')).toBe(true);
});

it('passes custom classnames', () => {
  const wrapper = shallow(
    <Button type="button" className="testClassName">
      Button
    </Button>,
  );

  expect(wrapper.find('button').hasClass('testClassName')).toBe(true);
});
