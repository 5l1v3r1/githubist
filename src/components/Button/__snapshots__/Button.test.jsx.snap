// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly primary button 1`] = `
<button
  className=""
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`renders correctly secondary button 1`] = `
<button
  className=""
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`renders correctly small button 1`] = `
<button
  className="undefined"
  disabled={false}
  type="button"
>
  Button
</button>
`;
